packages/agreement-process/src/services/agreementService.ts

```diff
@@ -658,12 +650,12 @@ export async function upgradeAgreementLogic(
     return {
       streamId: upgraded.id,
       events: [
-        toCreateEventAgreementUpdated(
+        toCreateEventAgreementArchivedByUpgrade(
```

Comment:
reminder: these events should be persisted using a single db transaction

Reply:
It seems that these events are still performed in two separated transaction.
Am I missing something?

Reply:
@MalpenZibo this point has not been addressed.
If this is intended to be in a separated PR, could you please open a task on Jira?

Reply:
https://pagopa.atlassian.net/browse/IMN-507